### ISSUE TITLE
Tune sklearn's neural networks

### ISSUE THREAD

**Issue by mfeurer** [automl/auto-sklearn#374]
As of sklearn 0.18 we have the chance to include neural networks into our configuration space. However, they are slow (b/c they're neural networks) and we need to think about how to set up the parametrization to allow the optimization to proceed in a reasonable amount of time.

**Comment by HamedMP** [comment]
Thinking about this issue, a few ideas came into reality:

1. For making an open source automl for neural networks, one can use Keras sklearn APIs to build even more complex models. Google AutoML is fine since they have TPUs and GPUs. This goal can be achieved in 1-2 years. Looking into [Using Machine Learning to Explore Neural Network Architecture](https://ai.googleblog.com/2017/05/using-machine-learning-to-explore.html) and [AutoML for large scale image classification and object detection](https://ai.googleblog.com/2017/11/automl-for-large-scale-image.html) can be helpful for this.

2. For short-term goals, we can have a look at NASnet and by assuming transfer learning is very successful for image recognition tasks, the code example in [NASnet in TF-SLIM](https://github.com/tensorflow/models/blob/master/research/slim/nets/nasnet/nasnet.py) or even [more complex apis ](https://github.com/tensorflow/models/tree/master/research/object_detection) can be used for retraining. As you can see in the first link, transfer learning itself has many parameters, and few layers can be added to the start (for normalization for example) and to the end of the model as MLP. Auto-sklearn then can be very useful for image processing tasks. I agree this is a bit off the main focus of auto-sklearn but seems very promising future on its own.

**Comment by franchuterivera** [comment]
Hello! I am closing this because it is an old issue. MLP classifier was already added to auto-sklearn
